components/Demo/Demo.tsx

```diff
@@ -10,63 +9,148 @@ import { TokenGateContext } from 'collabland-tokengate-react-context';
 import { useFormik } from 'formik';
 import { useContext } from 'react';
 import { useAccount } from 'wagmi';
+import * as Yup from 'yup';
+
+const defaultRules = {
+  type: 'ERC20',
+  chainId: 1, // Ethereum
+  minToken: '1',
+  contractAddress: '0xdac17f958d2ee523a2206206994597c13d831ec7', // Token Tether USD
+  roleId: '001',
+};
+
+const RulesValidationSchema = Yup.object().shape({
+  type: Yup.string().required('Required'),
+  chainId: Yup.number().required('Required'),
+  minToken: Yup.number().positive().required('Required'),
+  contractAddress: Yup.string().required('Required'),
+  roleId: Yup.string().required('Required'),
+});
 
 const Demo = () => {
-  const rules = [
-    {
-      type: 'ERC20',
-      chainId: 100,
-      minToken: '1',
-      contractAddress: '0x712b3d230F3C1c19db860d80619288b1F0BDd0Bd',
-      roleId: '001',
-    },
-  ];
   const { address } = useAccount();
+  const { checkRoles, isLoading, result, error } = useContext(TokenGateContext);
+
   const formik = useFormik({
     initialValues: {
       address: address,
+      chainId: defaultRules.chainId,
+      type: defaultRules.type,
+      contractAddress: defaultRules.contractAddress,
+      minToken: defaultRules.minToken,
+      roleId: defaultRules.roleId,
     },
     onSubmit: (values) => {
       checkRoles(
         {
-          account: values.address || '',
+          account: values.address!,
           rules: [
             {
-              type: 'ERC20',
-              chainId: 100,
-              minToken: '1',
-              contractAddress: '0x712b3d230F3C1c19db860d80619288b1F0BDd0Bd',
-              roleId: '001',
+              type: values.type!,
+              chainId: values.chainId!,
+              minToken: values.minToken!,
+              contractAddress: values.contractAddress!,
+              roleId: values.roleId,
             },
           ],
         },
-        process.env.NEXT_PUBLIC_COLLAB_LAND_API_KEY || ''
+        process.env.NEXT_PUBLIC_COLLAB_LAND_API_KEY!
       );
     },
+    validationSchema: RulesValidationSchema,
   });
-  const { checkRoles, isLoading, result, error } = useContext(TokenGateContext);
 
   return (
     <>
       <form onSubmit={formik.handleSubmit}>
-        <VStack spacing={4} align='flex-start'>
-          <FormControl>
-            <FormLabel htmlFor='address'>Address</FormLabel>
+        <VStack spacing={4} align="flex-start">
+          <FormControl isRequired>
+            <FormLabel htmlFor="address">Wallet Address</FormLabel>
             <Input
-              id='address'
-              name='address'
-              type='address'
-              variant='filled'
+              id="address"
+              name="address"
+              type="address"
+              placeholder="Enter Wallet Address"
               onChange={formik.handleChange}
               value={formik.values.address}
               disabled={isLoading}
+              variant="filled"
+              borderRadius={14}
```

Comment:
Border Radius, placeholder color and text color can be removed, I will fix that generically in theme ;) (Do this for all inputs)

Reply:
Fixed in [f6eca60](https://github.com/birimbau/token-gated-website-demo/pull/1/commits/f6eca60d844b49de2369d7b6eba3716fd76a530c)